### PR TITLE
Generate 'exist' rules based on foreign keys

### PR DESCRIPTION
**Pull request by Faryshta**

### CODE REVIEW

**Comment by Faryshta** [comment]
@samdark #56

File: generators/model/Generator.php [hunk 1]
```diff
@@ -317,6 +317,20 @@ public function generateRules($table)
             // doesn't support unique indexes information...do nothing
         }
 
+        // Exist rules for foreign keys.
+        foreach ($table->foreignKeys as $refs) {
+            $refClassName = $this->generateClassName($refs[0]);
+            unset($refs[0]);
+            $rules[] = "[['" . implode("', '", array_keys($refs)) . "'], "
+                . "'exist', "
+                . "'skypOnError' => true, "
```
Comment (samdark) [inline generators/model/Generator.php]: skyp → skip

**Comment by Faryshta** [comment]
what do I need to do to have this shipped with 2.0.5? @samdark ?

**Comment by cebe** [comment]
looks good to me.

**Comment by samdark** [comment]
Adjusted code https://github.com/yiisoft/yii2-gii/commit/4487b4600525f04b5e2e8a46593f04f62a9c3446 and merged. @Faryshta thanks!

**Comment by thiagotalma** [comment]
@samdark @cebe @Faryshta 
Guys, you need to check the need to use the brackets?

Here there was an error and found that the reason was the use of brackets when there is only one attribute.

The attribute is compared to NULL.

**Comment by samdark** [comment]
@thiagotalma can you elaborate?

**Comment by thiagotalma** [comment]
Generated rule:

```php
public function rules()
{
    return [
        [['client_id'], 'exist', 'skipOnError' => true, 'targetClass' => Client::className(), 'targetAttribute' => ['id']],
    ];
}
```

Query:
```sql
SELECT 1 FROM `pvw_Client` WHERE `id` IS NULL
```

Fixed rule:


```php
public function rules()
{
    return [
        ['client_id', 'exist', 'skipOnError' => true, 'targetClass' => Client::className(), 'targetAttribute' => 'id'],
    ];
}
```

Query:
```sql
SELECT 1 FROM `pvw_Client` WHERE `id`=135
```

**Comment by samdark** [comment]
https://github.com/yiisoft/yii2-gii/issues/64 ?

**Comment by Faryshta** [comment]
would

```php
public function rules()
{
    return [
        ['client_id', 'exist', 'skipOnError' => true, 'targetClass' => Client::className(), 'targetAttribute' => ['client_id' => 'id']],
    ];
}
```

work?

**Comment by thiagotalma** [comment]
@Faryshta same result (correct)

```sql
SELECT 1 FROM `pvw_Client` WHERE `id`=135
```

@samdark, not the same issue.

**Comment by samdark** [comment]
@thiagotalma, @Faryshta would you like to fix it?

**Comment by thiagotalma** [comment]
@Faryshta its your baby 😉

**Comment by cebe** [comment]
@thiagotalma can you open a new issue for that so we do not forget about it?

**Comment by Faryshta** [comment]
@thiagotalma @samdark @cebe i will fix it

**Comment by Faryshta** [comment]
#68 fix it and it passes the tests now.

**Pull request closed without merging**